os/arch/xtensa/src/esp32/esp_clk.h

```diff
@@ -0,0 +1,93 @@
+/******************************************************************
```

Comment:
rename `external/esp_idf_port/include/esp_clk.h` to `os/arch/xtensa/src/esp32/esp_clk.h` ?
or add a duplicated file?

Reply:
add a duplicated file. this PR mainly focus on remove the unreasonable dependency from os/arch to external,